sdk/tables/Azure.Data.Tables/tests/TableEntityTests.cs

```diff
@@ -178,5 +178,60 @@ public void DictionaryEntitySetNullProperties()
             entity[stringKey] = stringValue;
             Assert.That(entity[stringKey], Is.EqualTo(stringValue));
         }
+
+        [Test]
+        public void TypeCoercionForNumberTypes()
+        {
+            var entity = new TableEntity("partition", "row");
+
+            // Initialize a property to zero (int)
+            entity["IntToDouble"] = 0;
+            Assert.That(entity["IntToDouble"] is int);
```

Comment:
Should these also assert the expected value to ensure no loss of precision or unintentional nulling out?

Reply:
Good idea - added additional asserts